File: exchange-server/src/main/java/com/odmytrenko/model/ExchangeCurrencyRates.java
package com.odmytrenko.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.ToString;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.Table;
import java.util.UUID;

@JsonInclude(value = JsonInclude.Include.NON_EMPTY)
@Data
@ToString
@Entity
@Table(name = "CURRENCY_RATES")
@Inheritance(strategy = InheritanceType.JOINED)
public abstract class ExchangeCurrencyRates {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
        name = "UUID",
        strategy = "org.hibernate.id.UUIDGenerator"
    )
    private UUID id;
    private String bid;
    private String ask;
}

File: exchange-client/src/app/models/exchange-currency-rates.ts
export class ExchangeCurrencyRates {

  private _bid: string;
  private _ask: string;

  private _updated?: Date;
  private _bidChange?: string;
  private _askChange?: string;

  constructor({ bid, ask, updated, bidChange, askChange }: ExchangeCurrencyRates) {
    this.bid = bid.toString();
    this.ask = ask.toString();

    this.updated = updated;
    this.bidChange = bidChange.toString();
    this.askChange = askChange.toString();
  }

  get bid(): string {
    return this._bid;
  }

  set bid(value: string) {
    this._bid = value;
  }

  get ask(): string {
    return this._ask;
  }

  set ask(value: string) {
    this._ask = value;
  }

  get updated(): Date {
    return this._updated;
  }

  set updated(value: Date) {
    this._updated = value;
  }

  get bidChange(): string {
    return this._bidChange;
  }

  set bidChange(value: string) {
    this._bidChange = value;
  }

  get askChange(): string {
    return this._askChange;
  }

  set askChange(value: string) {
    this._askChange = value;
  }

}
